File: src/algebra1_two_step_equations_easy.py
"""
algebra1 - two_step_equations (easy)
Generated: 2026-02-22T03:47:49.571653
"""

from problem_utils import *

def generate():
    # For EASIER two-step equations (1000-1300 range), we want simple problems
    # Types: ax + b = c or x/a + b = c or ax - b = c
    
    problem_type = randint(1, 4)
    
    if problem_type == 1:
        # Type: ax + b = c (most basic two-step)
        a = nonzero(-5, 5)
        x_val = randint(-10, 10)
        b = nonzero(-10, 10)
        c = a * x_val + b
        
        equation = Eq(a*x + b, c)
        ans = x_val
        
        return problem(
            question=f"Solve for $x$: ${latex(a*x + b)} = {c}$",
            answer=ans,
            difficulty=(1000, 1200),
            topic="algebra1/two_step_equations",
            solution=steps(
                f"Given: ${latex(equation)}$",
                f"Subtract ${b}$ from both sides: ${latex(a*x)} = {c - b}$",
                f"Divide both sides by ${a}$: $x = {ans}$"
            ),
        )
    
    elif problem_type == 2:
        # Type: ax - b = c
        a = nonzero(-5, 5)
        x_val = randint(-10, 10)
        b = nonzero(1, 10)
        c = a * x_val - b
        
        equation = Eq(a*x - b, c)
        ans = x_val
        
        return problem(
            question=f"Solve for $x$: ${latex(a*x - b)} = {c}$",
            answer=ans,
            difficulty=(1000, 1200),
            topic="algebra1/two_step_equations",
            solution=steps(
                f"Given: ${latex(equation)}$",
                f"Add ${b}$ to both sides: ${latex(a*x)} = {c + b}$",
                f"Divide both sides by ${a}$: $x = {ans}$"
            ),
        )
    
    elif problem_type == 3:
        # Type: x/a + b = c (division version, easier numbers)
        a = choice([2, 3, 4, 5])
        x_val = a * randint(-8, 8)  # Make sure x is divisible by a
        b = randint(-10, 10)
        c = x_val // a + b
        
        equation = Eq(x/a + b, c)
        ans = x_val
        
        return problem(
            question=f"Solve for $x$: $\\frac{{x}}{{{a}}} + {b} = {c}$",
            answer=ans,
            difficulty=(1100, 1250),
            topic="algebra1/two_step_equations",
            solution=steps(
                f"Given: ${latex(equation)}$",
                f"Subtract ${b}$ from both sides: $\\frac{{x}}{{{a}}} = {c - b}$",
                f"Multiply both sides by ${a}$: $x = {ans}$"
            ),
        )
    
    else:
        # Type: b + ax = c (variable term second, slightly different)
        a = nonzero(-5, 5)
        x_val = randint(-10, 10)
        b = nonzero(-10, 10)
        c = b + a * x_val
        
        equation = Eq(b + a*x, c)
        ans = x_val
        
        return problem(
            question=f"Solve for $x$: ${b + latex(a*x)} = {c}$",
            answer=ans,
            difficulty=(1000, 1200),
            topic="algebra1/two_step_equations",
            solution=steps(
                f"Given: ${latex(equation)}$",
                f"Subtract ${b}$ from both sides: ${latex(a*x)} = {c - b}$",
                f"Divide both sides by ${a}$: $x = {ans}$"
            ),
        )

emit(generate())

File: src/algebra1_two_step_equations_easy.jl
# algebra1 - two_step_equations (easy)
# Generated: 2026-03-08T19:52:39.412935

include(joinpath(@__DIR__, "..", "julia", "src", "ProblemUtils.jl"))
using .ProblemUtils

@script x begin
    set_topic!("algebra1/two_step_equations")
    
    problem_type = rand(1:5)
    
    if problem_type == 1
        # Basic two-step: ax + b = c
        ans = randint(-15, 15)
        a = nonzero(-8, 8)
        b = randint(-20, 20)
        c = a * ans + b
        
        problem(
            question = "Solve for \\(x\\): \$$(a)x + $(b) = $(c)\$",
            answer = ans,
            difficulty = (700, 850),
            solution = steps(
                sol("Given", a*x + b ~ c),
                "Subtract $(b) from both sides: \$$(a)x = $(c - b)\$",
                "Divide both sides by $(a)",
                sol("Answer", x ~ ans)
            ),
            time = 60
        )
        
    elseif problem_type == 2
        # With fractions: x/a + b = c
        ans = randint(-20, 20)
        a = nonzero(-6, 6)
        b = randint(-15, 15)
        c = ans // a + b
        
        problem(
            question = "Solve for \\(x\\): \$\\frac{x}{$(a)} + $(b) = $(tex(c))\$",
            answer = ans,
            difficulty = (800, 950),
            solution = steps(
                sol("Given", x/a + b ~ c),
                "Subtract $(b) from both sides: \$\\frac{x}{$(a)} = $(tex(c - b))\$",
                "Multiply both sides by $(a)",
                sol("Answer", x ~ ans)
            ),
            time = 75
        )
        
    elseif problem_type == 3
        # With parentheses: a(x + b) = c
        ans = randint(-12, 12)
        a = nonzero(-7, 7)
        b = randint(-10, 10)
        c = a * (ans + b)
        
        problem(
            question = "Solve for \\(x\\): \$$(a)(x + $(b)) = $(c)\$",
            answer = ans,
            difficulty = (850, 1000),
            solution = steps(
                sol("Given", a*(x + b) ~ c),
                "Divide both sides by $(a): \$x + $(b) = $(c // a)\$",
                "Subtract $(b) from both sides",
                sol("Answer", x ~ ans)
            ),
            time = 75
        )
        
    elseif problem_type == 4
        # Distributive: a(x - b) + c = d
        ans = randint(-15, 15)
        a = nonzero(-6, 6)
        b = randint(-12, 12)
        c = randint(-18, 18)
        d = a * (ans - b) + c
        
        lhs = expand(a*(x - b) + c)
        
        problem(
            question = "Solve for \\(x\\): \$$(a)(x - $(b)) + $(c) = $(d)\$",
            answer = ans,
            difficulty = (900, 1100),
            solution = steps(
                sol("Given", a*(x - b) + c ~ d),
                "Distribute: \$$(tex(lhs)) = $(d)\$",
                "Subtract $(c) from both sides: \$$(a*x - a*b) = $(d - c)\$",
                "Add $(a*b) to both sides: \$$(a)x = $(d - c + a*b)\$",
                "Divide by $(a)",
                sol("Answer", x ~ ans)
            ),
            time = 90
        )
        
    else
        # Variables on both sides requiring two steps: ax + b = cx + d
        ans = randint(-18, 18)
        a = nonzero(-7, 7)
        b = randint(-25, 25)
        c = nonzero(-7, 7)
        while c == a
            c = nonzero(-7, 7)
        end
        d = (a - c) * ans + b
        
        problem(
            question = "Solve for \\(x\\): \$$(a)x + $(b) = $(c)x + $(d)\$",
            answer = ans,
            difficulty = (1000, 1200),
            solution = steps(
                sol("Given", a*x + b ~ c*x + d),
                "Subtract $(c)x from both sides: \$$(a - c)x + $(b) = $(d)\$",
                "Subtract $(b) from both sides: \$$(a - c)x = $(d - b)\$",
                "Divide by $(a - c)",
                sol("Answer", x ~ ans)
            ),
            time = 90
        )
    end
end